app/src/main/java/org/simple/clinic/navigation/v2/fragments/BaseScreen.kt

```diff
@@ -92,6 +92,8 @@ abstract class BaseScreen<K : ScreenKey, B : ViewBinding, M : Parcelable, E, F,
     disposable.add(events().subscribe { event ->
       _viewModel.dispatchEvent(event!!)
     })
+    
+    _viewModel.models.observe(viewLifecycleOwner, uiRenderer()::render)
```

Comment:
When it comes to method references, I think this pattern of calling a function and then using a reference on that seems a little unreadable. In this specific case, I'd personally prefer to extract the creation of the UI renderer to a separate variable.

```kotlin
val renderer = uiRenderer()
_viewModel.models.observe(viewLifecycleOwner, renderer::render)
```

What do you think?